ts/src/binance.ts

```diff
@@ -2855,10 +2878,21 @@ export default class binance extends Exchange {
         }
         const promises = await Promise.all (promisesRaw);
         let markets = [];
+        this.options['crossMarginPairsData'] = {};
+        this.options['isolatedMarginPairsData'] = {};
         for (let i = 0; i < fetchMarkets.length; i++) {
             const promise = this.safeDict (promises, i);
-            const promiseMarkets = this.safeList2 (promise, 'symbols', 'optionSymbols', []);
-            markets = this.arrayConcat (markets, promiseMarkets);
+            if (Array.isArray (promise)) {
```

Comment:
this is not correct, if we use safeDict it's not expected to check if it's an array 
![image](https://github.com/ccxt/ccxt/assets/43336371/ce7fea32-49ce-45e6-9ff4-eee7bbecfaaa)

Reply:
Also relying on the promise type dict vs array to infer the content of promise  seems a bit unsafe